tap_dynamodb/deserialize.py

```diff
@@ -58,23 +58,31 @@ def _apply_projection(self, record, breadcrumb, output):
                 breadcrumb_key = breadcrumb[0].split('[')[0]
                 index = int(breadcrumb[0].split('[')[1].split(']')[0])
                 if output.get(breadcrumb_key):
-                    output[breadcrumb_key].append(record[breadcrumb_key][index])
+                    # only prepare output if the list field contains data at that index position in record
+                    if len(record.get(breadcrumb_key)) > index:
```

Comment:
Can we merge this if with the parent if? Same for the else part as well.

Reply:
No, we cannot merge these conditions. When the original records contains `'metadata': ['test1']` and the projection we requested is `[['metadata[0]'], ['metadata[1]']]`.
As the **metadata** only contains 1 item, for the projection `['metadata[1]']` the output will become `[]` rather than `'metadata': ['test1']`.